### DISCUSSION TITLE
Convert Obsidian .md file to .tex LaTeX file and then directly compile it to PDF with pdflatex (trouble with the last step)

### DISCUSSION THREAD

**Discussion by ReaderGuy42** [Taitava/obsidian-shellcommands#360] [Q&A] [answered]
I have a Shell Command that I've been using for months:
```
mkdir -p /Obsidian/Output/{{title}} 
pandoc -r markdown-auto_identifiers -w latex {{file_path:absolute}} -o /Obsidian/Output/{{title}}/{{title}}.tex --template="s/LaTeX/My Templates/Paper/default.tex" --lua-filter=/Documents/LaTeX/Filters/highlight.lua --lua-filter=/Documents/LaTeX/Filters/highlight-period.lua --lua-filter=/Documents/LaTeX/Filters/Span.lua --lua-filter=/Documents/LaTeX/Filters/Span-period.lua & pdflatex {{title}}.tex
```
The filters are unrelated, I'm just leaving them in, otherwise I'll forgot them later. 
This script creates a folder in `Output` with the Obsidian file's name and converts it to .tex so I can then compile it in LaTeX. I just found out that I can directly compile a .tex file from Command Line with `pdflatex filename.tex`

However, this is where I'm having trouble. The little pop up in the top right corner keeps telling me that it's `Executing` my script but if I look in that folder there's nothing there but the .tex file. For reference, if I run it manually from the command line it does exactly what I want it to.

Any ideas?
Thanks!

**Comment by ReaderGuy42** [comment]
I've solved this like this:
```
mkdir -p /Documents/Obsidian/Output/{{title}} 
pandoc -r markdown-auto_identifiers -w latex {{file_path:absolute}} -o /Obsidian/Output/{{title}}/{{title}}.tex --template="/Documents/LaTeX/My Templates/Proposal/default-proposal.tex"
gnome-terminal -- bash -c "cd /Documents/Obsidian/Output/{{title}} && latexmk -pdf && latexmk -c && exit; exec bash"
```

This opens a terminal window, goes to that folder, runs latexmk and exits. Pretty happy with this! But if anyone knows how to then open the newly created PDF and then still exit the terminal without also closing the PDF reader, I'd be ecstatic.

I'm not having any luck with `nohup`. I've tried `latexmk -c && nohup xreader *.pdf && exit` and various other combinations, but that either opens the PDF reader, but also leaves the terminal open, closes both, or the whole script won't work lol.

**Reply by Taitava** [reply, marked as answer]
Hi. Sorry for the late reply!

First off, can this line...
> gnome-terminal -- bash -c "cd /Documents/Obsidian/Output/{{title}} && latexmk -pdf && latexmk -c && exit; exec bash"

... actually be simplified to this?
```
cd /Documents/Obsidian/Output/{{title}} && latexmk -pdf && latexmk -c
```

I don't know LaTeX or `latexmk` so I'm not able to provide support for them.

----------

> This opens a terminal window, goes to that folder, runs latexmk and exits. Pretty happy with this! But if anyone knows how to then open the newly created PDF and then still exit the terminal without also closing the PDF reader, I'd be ecstatic.

Have you tried this:
```
xdg-open *.pdf
```

**Reply by ReaderGuy42** [reply]
I've actually added some stuff and it now looks more complicated but it works. And I don't actually want it to open the PDF at all anymore, so it's fine :)
```
mkdir -p l/Documents/Obsidian/Output/Proposal/{{title}}
pandoc  -f markdown-blank_before_blockquote-auto_identifiers -w latex {{file_path:absolute}} -o /Documents/Obsidian/Output/{{title}}/{{title}}.tex --template="/Documents/LaTeX/My Templates/default.tex"
gnome-terminal -- bash -c "cd l/Documents/Obsidian/Output/Proposal/{{title}} && latexmk -pv- -pdflua < /dev/null && latexmk -c && cp {{title}}.pdf /Documents/Obsidian/Output/PDFs/ && exit; exec bash"
```

I switched the LaTeX compiler to LuaTeX because I need to be able to print some non-Latin characters. Thanks :)
